### PR TITLE
SSL Fix, install Fix, PHP 8 support vorbereitung, add setcookie Funct…

### PR DESCRIPTION
**Pull request by hhunderter**
…ion, Backend->Module List expansion

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds functionality)
- [X] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [X] This change requires a documentation update

# This PR has been tested in the following browsers:

- [X] Chrome
- [X] Firefox
- [X] Opera
- [X] Edge
- [X] IE

### CODE REVIEW

File: application/modules/admin/controllers/admin/Login.php [hunk 1]
```diff
@@ -87,14 +87,13 @@ public function logoutAction()
...
-            $params = session_get_cookie_params();
-            setcookie(session_name(), '', time() - 42000, $params['path'],
-                $params['domain'], $params['secure'], $params['httponly']
-            );
+            setcookieIlch(session_name(), '', strtotime('-12 hours'));
         }
 
         session_destroy();
 
+        session_destroy();
```
Comment (blackcoder87) [inline application/modules/admin/controllers/admin/Login.php:95]: session_destroy() gets called two times?

File: application/modules/checkoutbasic/config/config.php [hunk 2]
```diff
@@ -10,7 +10,7 @@ class Config extends \Ilch\Config\Install
 {
     public $config = [
         'key' => 'checkoutbasic',
-        'version' => '1.4.0',
+        'version' => '1.4.1',
```
Comment (blackcoder87) [inline application/modules/checkoutbasic/config/config.php:13]: Where is the change to the module that requires a higher version number?

File: application/libraries/Ilch/Functions.php [hunk 3]
```diff
@@ -471,3 +471,32 @@ function replaceVendorDirectory($tmpName = '_vendor')
 
     return false;
 }
+
+/**
+ * SetCookie Handling.
+ *
+ * @since 2.1.43
+ *
+ * @param string $tmpName temporary name of new vendor directory. Usually '_vendor'.
```
Comment (blackcoder87) [inline application/libraries/Ilch/Functions.php:480]: PHPDoc doesn't fit to the function.

- missing parameters
- wrong parameter names
- wrong description
- No return value

File: application/modules/checkoutbasic/config/config.php [hunk 4]
```diff
@@ -10,7 +10,7 @@ class Config extends \Ilch\Config\Install
...
         'icon_small' => 'fa-credit-card',
         'author' => 'Stantin, Thomas',
         'link' => 'http://ilch.de',
```
Comment (blackcoder87) [inline application/modules/checkoutbasic/config/config.php:16]: Could be updated to

`'link' => 'https://ilch.de',`

File: application/modules/checkoutbasic/config/config.php (hunk 2 above)
Comment (hhunderter) [reply application/modules/checkoutbasic/config/config.php:13]: Beim Update https://github.com/IlchCMS/Ilch-2.0/commit/19a57d64d7a122e93cd118aa1b3161d2806d6f87 wurde dies nicht angepasst

**Pull request merged**